### PR TITLE
Fix throwing unnecessary configurable compilation error

### LINKED ISSUE #41533: [Bug]: Confusing error messages when I use redeclared variable name for a `configurable` variable
### Description

I accidentally used an already declared variable name for a configurable variable. I couldn't understand the error messages initially. It took me a few minutes to figure out what was going on in VSCode.

### Steps to Reproduce

I defined the following variable without knowing that the same variable is defined somewhere else. 
```
configurable string host = ?;
``

I noticed red squiggly lines in VSCode and tried `bal build.` Here is the output:
```
ERROR [service.bal:(7:14,7:20)] invalid type for configurable variable: expected a subtype of 'anydata'
ERROR [service.bal:(7:21,7:25)] redeclared symbol 'host'
```


### Affected Version(s)

Ballerina Swan Lake Update 8.1

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject 

Fixes #41533 

## Approach
When there is a `redeclared symbol` error, the node type is changed to `other` at compilation. As the configurable validation only checks for type with `anydata`, it throws an unnecessary error. 
Modified the logic to early return if the type tag is `Semantic error`.

## Samples
```ballerina
string host = "ABC";
configurable string host = ?;
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by gabilang — APPROVED** [review]
LGTM

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java [hunk 1]
```diff
@@ -1156,9 +1156,11 @@ public void visit(BLangSimpleVariable varNode, AnalyzerData data) {
 
         BType lhsType = varNode.symbol.type;
         varNode.setBType(lhsType);
-
         // Configurable variable type must be a subtype of anydata.
         if (configurable && varNode.typeNode != null) {
+            if (lhsType.tag == TypeTags.SEMANTIC_ERROR) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java:1161]: Can we have one if condition here.

ex. if (configurable && varNode.typeNode != null && lhsType.tag != TypeTags.SEMANTIC_ERROR)

Comment (warunalakshitha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java:1161]: Do we need to chck varNode.typeNode != null

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**